pkg/network/protocols/events/worker_pool.go

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
> why not taking a ready implementation of a worker pool? 
I don't have a good answer for that to be honest. It seemed like a simple enough thing to implement myself, but I definitely think there are always pros and cons in doing so.

As to your second point, I actually _don't think_ we should handle panics inside the pool library. we certainly don't want a scenario where you have a dead pool of workers and you "silently" block indefinitely when enqueuing a job, but in our case a panic would bubble up to all the way up to the main goroutine and the program would crash, which is totally acceptable behavior IMO (the summarize, it's almost always a bad idea to handle panics which is why Go doesn't even have exceptions and doesn't give you much to deal with panics :sweat_smile: )

Reply:
I go get the second answer
But let's revisit the first bullet
IMHO, if we don't have any good reason and if there are good OSS candidates, we should choose one and use it
I want to maintain less code whenever it is possible

Reply:
I definitely want to avoid reinventing the wheel where possible, but I also think we need to be judicious when deciding to use an external library. 
My thinking is, we're responsible for the code we ship to customers whether it is our own or from a third party library. By picking an external library we're not just offloading the responsibility of maintaining code, we're vetting it and becoming co-responsible for it (not to mention things like licensing, security vulnerabilities etc)
I took a cursory glance at a couple of libraries from the list you linked and frankly they felt a bit overkill for the most part.
Given that we're talking about 30 LOC with a very specific purpose and a single use case, I just don't think it's worth it